### PR TITLE
Core options ifdefs

### PR DESCRIPTION
**Pull request by dmrlawson**
I think I've got all of the for the main options. Someone should double check the logic is correct.

There seems to be a half followed convention of commenting what the `#endif` is about, shall I add that or not?

### CODE REVIEW

File: libretro/libretro_core_options.h [hunk 1]
```diff
@@ -224,7 +237,11 @@ struct retro_core_option_definition option_defs_us[] = {
             {"Sync", NULL},
             {NULL, NULL},
         },
+#ifndef HAVE_OPENGLES
         "Async"
```
Comment (dankcushions) [inline libretro/libretro_core_options.h:241]: I think this logic is slightly wrong - see https://github.com/libretro/mupen64plus-libretro-nx/pull/160/files

File: libretro/libretro_core_options.h [hunk 2]
```diff
@@ -304,8 +325,13 @@ struct retro_core_option_definition option_defs_us[] = {
             {"True", NULL},
             {NULL, NULL},
         },
+#ifdef HAVE_OPENGLES
+        "False
+#else
         "True"
```
Comment (dankcushions) [inline libretro/libretro_core_options.h:331]: this is enabled by default on upstream - https://github.com/gonetz/GLideN64/blob/a96853d559680cca17fdc4dfa5891dad03d79b66/src/Config.cpp#L58 - safe? works on my GLES2 and 3 devices

File: libretro/libretro_core_options.h (hunk 1 above)
Comment (dmrlawson) [reply libretro/libretro_core_options.h:241]: This is just copied from here: https://github.com/libretro/mupen64plus-libretro-nx/blob/develop/libretro/libretro.c#L274

I would suggest that this MR simply replicates the current state and the other merge request deals with the behaviour change, but I'm open to doing it here if that's what's desired.

File: libretro/libretro_core_options.h (hunk 2 above)
Comment (dmrlawson) [reply libretro/libretro_core_options.h:331]: This is from https://github.com/libretro/mupen64plus-libretro-nx/blob/develop/libretro/libretro.c#L293-L294

Again I'd suggest a separate MR for changing behaviour.

**Comment by dmrlawson** [comment]
Also, the `EnableNoiseEmulation` option needs to be removed as it's not configurable in the latest GLideN64, but I left it for now as this is to merge into the `improved_coreoptions` branch which doesn't have it yet

**Comment by dankcushions** [comment]
oh sorry, didn't realise this wasn't for main branch 👶 
we can revisit this stuff later

**Comment by m4xw** [comment]
We will Incooperate your feedback, dw.
didn't forget about it

**Comment by m4xw** [comment]
CXD4 and Parallel are opt in https://github.com/libretro/mupen64plus-libretro-nx/blob/develop/libretro/libretro.c#L244-L246
https://github.com/libretro/mupen64plus-libretro-nx/blob/develop/libretro/libretro.c#L95-L116
Using some macro trickery.
https://github.com/libretro/mupen64plus-libretro-nx/blob/develop/libretro/libretro.c#L265-L270
https://github.com/libretro/mupen64plus-libretro-nx/blob/develop/libretro/libretro.c#L345-L354

**Comment by dmrlawson** [comment]
Fixed. Got rid of the macros because I don't think they're necessary now

**Comment by m4xw** [comment]
Gonna merge for now and do manual fixup if required.

**Pull request merged**